x-pack/legacy/plugins/infra/server/lib/alerting/metric_threshold/register_metric_threshold_alert_type.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { schema } from '@kbn/config-schema';
+import { Server } from 'hapi';
+import {
+  MetricThresholdAlertTypeParams,
+  Comparator,
+  AlertStates,
+  METRIC_THRESHOLD_ALERT_TYPE_ID,
+} from './types';
+import { AlertServices } from '../../../../../alerting/server/types';
+
+const FIRED_ACTIONS = {
+  id: 'metrics.threshold.fired',
+  name: i18n.translate('xpack.metrics.alerting.threshold.fired', {
+    defaultMessage: 'Fired',
+  }),
+};
+
+async function getMetric(
+  { callCluster }: AlertServices,
+  { metric, searchField, aggregation, interval, indexPattern }: MetricThresholdAlertTypeParams
+) {
+  const searchBody = {
+    query: {
+      bool: {
+        filter: [
+          {
+            range: {
+              '@timestamp': {
+                gte: `now-${interval}`,
+              },
+            },
+          },
+          {
+            bool: {
+              should: [
+                {
+                  match_phrase: {
+                    [searchField.name]: searchField.value,
+                  },
+                },
+                {
+                  exists: {
+                    field: metric,
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    },
+    size: 0,
+    aggs: {
+      aggregatedIntervals: {
+        date_histogram: {
+          field: '@timestamp',
+          calendar_interval: interval,
+        },
+        aggregations: {
+          aggregatedValue: {
+            [aggregation]: {
+              field: metric,
+            },
+          },
+        },
+      },
+    },
+  };
+
+  const result = await callCluster('search', {
+    body: searchBody,
+    index: indexPattern,
+  });
+
+  const { buckets } = result.aggregations.aggregatedIntervals;
+  const { value } = buckets[buckets.length - 1].aggregatedValue;
```

Comment:
Since you're only using the last bucket, the date histogram doesn't seem to be that useful here.  But this function could be repurposed to deal with a "history" of what the alert would have done with old data, eg, for usage in a preview UI.  We're doing something similar with the not-yet-merged built-in alerting index threshold alertType.

Reply:
I'm still an ES query newbie, what approach would you recommend for getting this data without using histogram buckets?

Reply:
I believe you can use a simple "metrics" aggregation, which doesn't do any bucketing, eg https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-avg-aggregation.html .  You've already got the set of documents to use to generate the metric value you want via the filter, and you don't need those documents further bucketed.  That's what I'd try, but I'd also have someone else review it, as it wouldn't surprise me if there was something even easier - because I am also an ES query n00b :-).  I actually kinda wonder if an esSQL query would be the easiest thing here (and thinking about it a little for the index threshold as well).